shared/api/tokenizer_impl.cc

```diff
@@ -1,41 +1,61 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+
 #include "bpe_kernels.h"
 #include "bpe_tokenizer.hpp"
 #include "bpe_decoder.hpp"
+#include "ugm_kernels.hpp"
 
 #include "tokenizer_impl.h"
 
-using namespace ort_extensions;
 
-TokenizerImpl::TokenizerImpl() : OrtxObjectImpl(extObjectKind_t::kOrtxKindTokenizer) {};
+namespace ort_extensions {
+
+TokenizerImpl::TokenizerImpl()
+    : OrtxObjectImpl(extObjectKind_t::kOrtxKindTokenizer) {};
 TokenizerImpl::~TokenizerImpl() {};
 
-OrtxStatus TokenizerImpl::Load(const std::string& dir) {
+OrtxStatus TokenizerImpl::Load(const std::string& tok_path) {
+  // bool is_tiktoken = false;
+  // ortx::path tok_path_obj(tok_path);
+  // if (tok_path_obj.is_regular_file() && tok_path_obj.extension() == ".tiktoken") {
+  //   is_tiktoken = true;
+  // }
+
+  // std::string tok_dir = tok_path;
+  // if (tok_path_obj.is_regular_file()) {
+  //   tok_dir = tok_path_obj.parent_path();
+  // }
+
   tok_config_ = std::make_shared<ort_extensions::bpe::TokenJsonConfig>();
-  auto status = tok_config_->Load(dir);
+  auto status = tok_config_->Load(tok_path);
   if (!status.IsOk()) {
     return status;
   }
 
-  auto vocab_file_path = path(dir) / "tokenizer.json";
-  std::ifstream vocab_fs = vocab_file_path.open();
+  if (tok_config_->tokenizer_class_.empty()) {
+    auto tokenizer = std::make_unique<SpmUgmTokenizer>();
+    status = tokenizer->Load(*tok_config_);
+    if (status.IsOk()) {
+      tokenizer_ = std::move(tokenizer);
+    }
 
-  tokenizer_ = std::make_unique<JsonFastTokenizer>();
-  if (!vocab_fs.is_open()) {
-    // No tokenizer.json file present; use TikToken tokenizer
-    tokenizer_->tiktoken_ = true;
+    return status;
+  }
 
-    // load the tokenizer from a config
-    status = tokenizer_->Load(*tok_config_);
-  } else {
-    // load the tokenizer from a config
-    status = tokenizer_->Load(*tok_config_);
+  auto vocab_file_path = ortx::path(tok_config_->GetVocabDataFile());
+  auto tokenizer = std::make_unique<JsonFastTokenizer>();
+  auto fx_load = vocab_file_path.extension() == ".json"?
```

Comment:
perhaps we can also have a check here for if it's neither - so if `vocab_file_path.extension()` is ".json" then we use `JsonFastTokenizer::Load`, else if it's ".tiktoken", we use `JsonFastTokenizer::LoadTikTokenBase64`, otherwise we throw and error message saying something like "Tokenizer file type not supported! Please use .json or .tiktoken files to load tokenizer."

that way, if the user or anyone has errors in the future we immediately know what the root cause is.

Reply:
the vocab file path should be checked in JsonConfig load which is the center place to do all checks. Add a comment here for calarifiction.